### PR TITLE
#162949210 Setup jwt authentication

### PR DESCRIPTION
**Pull request by kwanj-k**
#### What does this PR do?

Configure jwt authentication and tests for the same

#### Description of Task to be completed?

* Write tests to check if the token is returned on login/registration
* Create a token on registration/signup

#### How should this be manually tested?

1. Clone the repo: `git clone https://github.com/andela/ah-the-unsullied.git`
2. Cd into the cloned repo 
3. Git checkout to branch ft-configure-jwt-162949210
4. Set up a virtual environment and install requirements:
      ```
            python3 -m venv venv
            source venv/bin/activate
            pip3 install -r requirements.txt

     ```
5. Set up a .env file in the same level as the venv file and add the following:
      ```
            source venv/bin/activate
            export SECRET_KEY='yoursecretkey'
             export DATABASE_URL='postgres://yourdatabasename'

      ```
6. Then run: 
     ``` 
            source .env 
            ./manage.py migrate
            ./manage.py runserver
      ```

7. Fire up postman and test the following endpoints:

| EndPoint                       | Functionality                           |
| ---------------------- |:----------------------------:|
| POST     api/users/      | Register User                          |
| POST     api/users/login/ | Login a user                        |

                                                    
use the following payload format:
```
{
  "user": {
    "email": "jake@jake.jake",
    "username": "jake",
    "password":"password"
  }
}
```

#### Any background context you want to provide?
None
#### What are the relevant pivotal tracker stories?
[162949210](#162949210)
#### Screenshots (if appropriate)
None
#### Questions:

### CODE REVIEW

File: authors/apps/authentication/backends.py [hunk 1]
```diff
@@ -1,10 +1,21 @@
...
-# from .models import User
+"""
+JWT configuration
+"""
+from rest_framework_jwt.settings import api_settings
 
-"""Configure JWT Here"""
+#By default, django-restframework--jwt generates a token 
+#here we create a function to manually create a token
+#with user data of our choosing
```
Comment (indungu) [inline authors/apps/authentication/backends.py]: Please use the PEP8 multiline comment style guides.

**Review by Allan690 — DISMISSED** [review]

**Review by MbugwaSami — DISMISSED** [review]

File: authors/apps/authentication/backends.py [hunk 2]
```diff
@@ -1,10 +1,21 @@
...
-# from rest_framework import authentication, exceptions
-#
-# from .models import User
+"""
+JWT configuration
+"""
+from rest_framework_jwt.settings import api_settings
 
-"""Configure JWT Here"""
+#By default, django-restframework--jwt generates a token
```
Comment (nesh-dev) [inline authors/apps/authentication/backends.py]: To follow PEP8 standards there should be space between the # and the start of the comment.
There should also be space between the comment and the function.

**Review by kelvinwm — DISMISSED** [review]

File: authors/apps/authentication/serializers.py [hunk 3]
```diff
@@ -15,17 +16,21 @@ class RegistrationSerializer(serializers.ModelSerializer):
...
 
     class Meta:
         model = User
         # List all of the fields that could possibly be included in a request
         # or response, including fields specified explicitly above.
-        fields = ['email', 'username', 'password']
+        fields = ['email', 'username','password','token']
 
     def create(self, validated_data):
+        #print(validated_data)
```
Comment (nesh-dev) [inline authors/apps/authentication/serializers.py]: kindly remove the commented lines.

**Review by nesh-dev — CHANGES_REQUESTED** [review]

**Review by nesh-dev — DISMISSED** [review]

**Review by sylviawanjiku — DISMISSED** [review]

**Review by nesh-dev — DISMISSED** [review]

**Review by sylviawanjiku — DISMISSED** [review]

File: authors/apps/authentication/backends.py [hunk 4]
```diff
@@ -1,10 +1,22 @@
...
+# By default, django-restframework--jwt generates a token 
+# here we create a function to manually create a token
+# with user data of our choosing
+# Here the token is passed the user email
+# All the the jwt configurations are defined globally in the project settings.py
 
+def get_jwt_token(user):
+    """
+    Creates a token manually
+
```
Comment (indungu) [inline authors/apps/authentication/backends.py]: Please remove unnecessary whitespace characters.

**Review by indungu — CHANGES_REQUESTED** [review]

**Review by sylviawanjiku — DISMISSED** [review]

**Review by nesh-dev — DISMISSED** [review]

**Review by Allan690 — DISMISSED** [review]

**Review by sylviawanjiku — APPROVED** [review]

**Review by nesh-dev — APPROVED** [review]

**Review by MbugwaSami — APPROVED** [review]

**Review by indungu — APPROVED** [review]

**Pull request merged**